### ISSUE TITLE
When using Filter for base64_encode on a string with dollar signs, encoded base64 is cut off

### ISSUE THREAD

**Issue by ccben87** [Shuffle/Shuffle#1454]
**Describe the bug**

When using the Http app and using the curl function, when using "Expand window" and applying a base64_encode filter on a string with $ in it I get a syntax error (this is probably correct and by design): "Errors (1): There was a syntax error in your Liquid input (2). Details: expected token 'end of print statement', got ':'"

So I wrap the string in quotes, and then it is allowed as valid syntax and the "TRY IT" button then shows that it works (it encodes into correct base64) but when actually executing the query again curl

Example:

Input into the "Expand window" dialog box: `curl -H "Authorization: Basic {{"username:tResW0jEd-*Pr4W-mocU$6ra-437LtR*4ispuStAc" | base64_encode }}" -k -X GET "https://<snip>"`

Hit TRY IT and it comes out as (which is correct encoded base64): `curl -H "Authorization: Basic dXNlcm5hbWU6dFJlc1cwakVkLSpQcjRXLW1vY1UkNnJhLTQzN0x0Uio0aXNwdVN0QWM=" -k -X GET "https://<snip>"`

![image](https://github.com/user-attachments/assets/7a038787-3f9f-4bb6-9274-e6ae62071b03)

However, when going to execute the curl with the http app: `curl -H "Authorization: Basic dXNlcm5hbWU6dFJlc1cwakVkLSpQcjRXLW1vY1UqNGlzcHVTdEFj" -k -X GET "https://<snip>"`

![image](https://github.com/user-attachments/assets/a835842c-e49a-4ac2-a983-d4907bf4b227)

As you can see, the base64 ends up being incorrectly encoded when actually running the app, it's being cut off at the $ sign.

A clear and concise description of what the bug is. If you can't execute a workflow, or an app isn't being created, an app isn't being created etc, please add debug logs (see further down)

**To Reproduce**
1. Create new workflow
2. Add "Http" app
3. Configure "Http" app to use curl
4. Enter a statement with a base64_encode filter containing dollar signs eg. {{ "TEXTHERE$$$$INCLUDESDOLLARSIGNS" |  base64_encode }}
5. Run the app
6. Find that the base64 output is cut off at the dollar signs.
...

**Expected behavior**
There needs to be a way to include dollar signs when using these "Filters", the execution result should match the result for the test output.

**Comment by frikky** [comment]
Hey! You have to escape dollarsigns as they are the only custom character we manage

$ -> \$

**Comment by ccben87** [comment]
> Hey! You have to escape dollarsigns as they are the only custom character we manage
Ah, thanks. That is helpful. That certainly works. However, I think it would be best long term if this was reflected in the test output so as not to confuse.

Also, more importantly in my mind, this has shown me that when using the Basic Authentication or an App Auth setting in Shuffle that passwords that include dollar signs also need to be escaped! Now that I understand that dollar signs need to be escaped, my authentications work without issue. In my view when entering a password into such fields that Shuffle should automatically escape this as necessary. It's certainly not very intuitive to copy and paste a password in here, to not even notice that it happens to have a dollar sign in it (due to the password masking), only to wonder why the auth keeps failing:

![image](https://github.com/user-attachments/assets/7a48d58a-50fb-4265-b796-203f45514466)

Thanks!

**Comment by frikky** [comment]
> > Hey! You have to escape dollarsigns as they are the only custom character we manage
> 
> Ah, thanks. That is helpful. That certainly works. However, I think it would be best long term if this was reflected in the test output so as not to confuse.
> 
> Also, more importantly in my mind, this has shown me that when using the Basic Authentication or an App Auth setting in Shuffle that passwords that include dollar signs also need to be escaped! Now that I understand that dollar signs need to be escaped, my authentications work without issue. In my view when entering a password into such fields that Shuffle should automatically escape this as necessary. It's certainly not very intuitive to copy and paste a password in here, to not even notice that it happens to have a dollar sign in it (due to the password masking), only to wonder why the auth keeps failing:
> 
> ![image](https://private-user-images.githubusercontent.com/128038118/350207647-7a48d58a-50fb-4265-b796-203f45514466.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEzOTAzOTQsIm5iZiI6MTcyMTM5MDA5NCwicGF0aCI6Ii8xMjgwMzgxMTgvMzUwMjA3NjQ3LTdhNDhkNThhLTUwZmItNDI2NS1iNzk2LTIwM2Y0NTUxNDQ2Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxOVQxMTU0NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNDU4ZGIwYWNhZDcyOTgxZGQyMDBmNjZlNzc0ZmUzNmMxODAxMGNmY2M2MmE2M2JmODI2OTE1NGEyMTI5Nzk4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.cz7ALtkXTK9gPqMl_5HgLVTOWpEjZke54Hfx2V2GbFI)
> 
> Thanks!

You are right that there is a bad pattern here, and we appreciate the input. The reason it works like this is because you can use references even within authentication that uses $ references. Having it work otherwise would sadly cause bunch of edgecases to happen :(

**Comment by ccben87** [comment]
> The reason it works like this is because you can use references even within authentication that uses $ references. Having it work otherwise would sadly cause bunch of edgecases to happen :(

Oh, that's a shame :(. Could we then please potentially have a little textual warning below, perhaps dynamically appearing when it is detected that there is a '$' entered? I believe this would make it a lot more simple for new users. With the way it is now, it could cause frustration and perhaps cause a new user to give up and decide not to use Shuffle.

**Comment by frikky** [comment]
> > The reason it works like this is because you can use references even within authentication that uses $ references. Having it work otherwise would sadly cause bunch of edgecases to happen :(
> 
> Oh, that's a shame :(. Could we then please potentially have a little textual warning below, perhaps dynamically appearing when it is detected that there is a '$' entered? I believe this would make it a lot more simple for new users. With the way it is now, it could cause frustration and perhaps cause a new user to give up and decide not to use Shuffle.

That's a good idea! 

@Monilprajapati can you add an error for a parameter that has parameter.configuration=true to show a warning if they type an unescaped dollarsign? 

Shows warning:
`sometextx$afterthetext`

Doesn't show warning:
`sometextx\$afterthetext`

**Comment by Monilprajapati** [comment]
> > > The reason it works like this is because you can use references even within authentication that uses $ references. Having it work otherwise would sadly cause bunch of edgecases to happen :(
> > 
> > 
> > Oh, that's a shame :(. Could we then please potentially have a little textual warning below, perhaps dynamically appearing when it is detected that there is a '$' entered? I believe this would make it a lot more simple for new users. With the way it is now, it could cause frustration and perhaps cause a new user to give up and decide not to use Shuffle.
> 
> That's a good idea!
> 
> @Monilprajapati can you add an error for a parameter that has parameter.configuration=true to show a warning if they type an unescaped dollarsign?
> 
> Shows warning: `sometextx$afterthetext`
> 
> Doesn't show warning: `sometextx\$afterthetext`
> 
> Also add this to the Authentication popup where the put in text.

Added in this #1460 PR
